### PR TITLE
feat: track env vars from templates

### LINKED ISSUE #142: Extract Env var references in test suite and code into a common const set

### PR DESCRIPTION
**Pull request by bartoszmajsak**
#### Short description of what this resolves:

Collects env vars/parameters defined in templates so we can document them easily in the CLI and 

#### Changes proposed in this pull request:

- adds a description to parameters in templates
- extracts env var string to consts
- collect tpl parameters

Fixes #142 

NOTE: help string and actual usage will come as part of #117

### CODE REVIEW

File: cmd/ike/config/env_vars.go [hunk 1]
```diff
@@ -0,0 +1,38 @@
...
+	EnvDockerRegistry   = "IKE_DOCKER_REGISTRY"
+)
+
+var (
+	log        = logf.Log.WithName("session_handler")
+	parameters []openshiftApi.Parameter
+)
+
+func init() {
+	collectTplParams("operator.yaml")
```
Comment (aslakknutsen) [inline cmd/ike/config/env_vars.go]: For future safety; iterate over all Assets?

File: pkg/openshift/parser/template_test.go [hunk 2]
```diff
@@ -113,22 +113,27 @@ var _ = Describe("template processing", func() {
 	})
 })
 
-const processedTmplWithDefaults = `kind: Template
+const processedOperatorTmplWithDefaults = `kind: Template
```
Comment (aslakknutsen) [inline pkg/openshift/parser/template_test.go]: Any reason to keep this vs reading the actual file?

File: test/cmd/test-scenario/main.go [hunk 3]
```diff
@@ -37,10 +39,10 @@ func getTestImageName() string {
...
 
-	if s, f := os.LookupEnv("IKE_DOCKER_REGISTRY"); f {
+	if s, f := os.LookupEnv(config.EnvDockerRegistry); f {
 		reg = s + "/"
 	}
-	if s, f := os.LookupEnv("IKE_DOCKER_REPOSITORY"); f {
+	if s, f := os.LookupEnv(config.EnvDockerRepository); f {
 		repo = s
 	}
 	if s, f := os.LookupEnv("IKE_TEST_IMAGE_NAME"); f {
```
Comment (aslakknutsen) [inline test/cmd/test-scenario/main.go]: Add to config even tho not in Template?

File: pkg/openshift/parser/template_test.go (hunk 2 above)
Comment (bartoszmajsak) [reply pkg/openshift/parser/template_test.go]: This one is processed - so have values replaced.

File: test/cmd/test-scenario/main.go (hunk 3 above)
Comment (bartoszmajsak) [reply test/cmd/test-scenario/main.go]: as `const`?

Comment (bartoszmajsak) [reply test/cmd/test-scenario/main.go]: 1177045

File: cmd/ike/config/env_vars.go (hunk 1 above)
Comment (bartoszmajsak) [reply cmd/ike/config/env_vars.go]: Ace! ace0ee7

**Pull request merged**